tiledb/sm/serialization/fragment_info.cc

```diff
@@ -249,6 +249,7 @@ single_fragment_info_from_capnp(
       single_frag_info_reader.getFragmentSize(),
       meta->non_empty_domain(),
       expanded_non_empty_domain,
+      meta->shape_data(),
```

Comment:
This means you need to add serialization/deserialization for `shape_data_` in `FragmentMetadata` capnp or this will just be empty/default initialized... Is that intentional ? Same probably for `has_shape_data_` that controls that value if I understand correctly.

Reply:
Adding checks for `shape_data` to the serialization tests in `test/src/unit-cppapi-fragment_info.cc` could help verify.